parser/prism/Translator.cc

```diff
@@ -76,6 +76,71 @@ template <typename... TArgs>
     Exception::raise(reasonFormatStr, forward<TArgs>(args)...);
 }
 
+// Helper function to check if an AST expression is a string literal
+bool isStringLit(ast::ExpressionPtr &expr) {
+    if (auto lit = ast::cast_tree<ast::Literal>(expr)) {
```

Comment:
```suggestion
    if (auto *lit = ast::cast_tree<ast::Literal>(expr)) {
```

Rationale: https://github.com/sorbet/sorbet/pull/9235#discussion_r2308636244